odoo/addons/base/res/res_partner.py

```diff
@@ -624,6 +624,11 @@ def name_create(self, name):
             If only an email address is received and that the regex cannot find
             a name, the name will have the email value.
             If 'force_email' key in context: must find the email address. """
+        context = dict(self._context)
+        default_type = context.get('default_type', False)
```

Comment:
No need to specify False, default value of get is None and we care only about not falsy values here.